src/main/scala/io/github/sentenza/hacktoberfest18/algos/MutableSorting.scala

```diff
@@ -185,10 +188,41 @@ package io.github.sentenza.hacktoberfest18.algos
       array
     }
 
-    def bucketSort(xs: Array[Int]): Array[Int] = ???
+  /**
+    * @author Xoeseko
+    * Bucket sort is a sorting algorithm that sorts an array of elements by splitting the elements into n buckets
+    * and then reapplies another sorting method on the different buckets. Then merges the sorted buckets.
+    * It can also be used recursively until buckets contain one element each.
+    *
+    * @param xs Array of sortable integers
+    * @param n number of buckets in which to sort the elements.
+    * @param sort The sorting algorithm to apply once buckets are divided.
+    * @return the sorted array
+    */
+  def bucketSort(xs: Array[Int], n: Int = 10, sort: Array[Int] => Array[Int] = insertionSort): Array[Int] = {
+    val buckets = new Array[Array[Int]](n)
+    val temp: ArrayBuffer[Int] = new ArrayBuffer[Int]()
```

Comment:
you can simply 

```scala
val temp = ArrayBuffer[Int]()
```

Reply:
I actually like my explicit types... Especially for my array buckets making it impossible to assign a simple array of integers to the buckets... I think it makes the method more robust I actually should add a type to buckets... Why would you suggest simplifying the syntax @ivanopagano simply for readability ? Or do you really think they don't bring much in this case ?

Reply:
What @ivanopagano was trying to point out is the fact that through type inference, the Scala complier is smart enough to figure out the types for you, and moreover once the `val temp` is defined is impossible to re-assign it (because its immutability).

Reply:
True I didn't think about the fact a value's type could be inferred and would then be immutable ! thanks

Reply:
@Xoeseko welcome to static typing land :smile: